sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1561,6 +1561,12 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       messageParameters = Map("changes" -> changes.toString()))
   }
 
+  def hintUnsupportedForJdbcDatabasesError(jdbcDialect: String): AnalysisException = {
+    new AnalysisException(
+      errorClass = "HINT_UNSUPPORTED_FOR_JDBC_DIALECT",
+      messageParameters = Map("jdbcDialect" -> jdbcDialect))
+  }
```

Comment:
The error is not belongs the scope of compilation.
Please use `QueryExecutionErrors`.